opentelemetry-api/src/opentelemetry/trace/status.py

```diff
@@ -183,3 +183,25 @@ def description(self) -> typing.Optional[str]:
     def is_ok(self) -> bool:
         """Returns false if this represents an error, true otherwise."""
         return self._canonical_code is StatusCanonicalCode.OK
+
+
+class StatusError(Exception):
```

Comment:
Ok, but without a status-specific exception like this one, how can the auto-status-update mechanism know which status to set? @lzchen mentions [here](https://github.com/open-telemetry/opentelemetry-python/issues/292#issue-523580208): _In OpenCensus the status seems to be derived from the exception thrown._

This PR tries to set the status depending on the exception thrown by simply taking the status from the exception and using it to set the status. That, of course, makes it necessary that the user raises a `StatusError` with the desired status in their application code. Just to be sure that I understand you correctly, is that necessity what you consider a mix of observability code with application logic?

I wonder if we can work around this problem by making the context manager "smart" in some way, with that I mean capable of setting the status without being explicitly informed which status to set by the raised exception. I'll try to come up with some idea, please let me know your thoughts about this too, @Oberon00.

Reply:
> how can the auto-status-update mechanism know which status to set

I think it would be nice to have an API like `addExceptionToStatusTranslator(translator: Callable[[Exception], Status]` that returns `None` if it doesn't know the exception and otherwise a Status for the exception. These could then be queried in order. But I also think this is beyond the scope of this PR and for now just always using Unknown is fine.

Reply:
Ok, I have updated the PR with the removal of `StatusError`. After taking a look at OpenCensus [here](https://github.com/census-instrumentation/opencensus-python/blob/e9129b74681df204c027702360be20fc0eea7f76/opencensus/trace/status.py#L73-L78) I realized that they basically just set the status to Unknown. This matches what you have suggested above. I implemented like this, so basically, the behavior is equivalent to OpenCensus on this matter.

I kept the removal of `StatusError` in a separate commit, just in case. I'll squash when this has your LGTM, @Oberon00.

Now, regarding the mixture of observability and application code, I have just a slight concern. I think that you have a valid point when you mention that this is not desired, but then you mention that: _you can always just set the span status directly instead of raising a special exception_

Isn't setting the status directly *also* a mix of observability and application code?

The previously implemented mechanism allowed the user to raise an exception while also setting the status, if desired. Without this, I guess the user will set the status directly and then raise an exception. I find these two approaches kind of equivalent, both apparently mix observability and application code.

So, if we want to avoid this mix, should we also make the status setting method at least not part of the interface that the user has access to?

Well, this is quickly getting in the realm of design philosophy :smile: Just for the record, and to get things moving, I am ok with merging your requested changes that are now implemented. Just wanted to know your thoughts on this issue of mixing application and observability code. :+1: 

Thank you, @Oberon00!